os/drivers/compression/compress.c

```diff
@@ -193,7 +193,7 @@ static int comp_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
 		ret = data->compression_header->binary_size;
 		if (ret <= 0) {
 			bcmpdbg("Failed to get buffer size = %d\n", ret);
-			ret = ERROR;
+			return -EINVAL;
```

Comment:
-EINVAL means that argument is wrong.
But this error means that file doesn't include compression header in its priv data area.
Hence I think EBADF(bad file descriptor) is most proper errno